README.md

```diff
@@ -4,6 +4,39 @@
 
 Tools for implementing SCSI on top of NVMe
 
+This repo contains the following:
+1) SCSI mock device
+1) NVMe device driver
+1) Translation Library
+1) Engine to orchestrate execution flow
+
+## End-to-end translation ##
+
+### Setup ###
+1) Ensure the following:
+    - an NVMe device is attached to the machine at /dev/nvme0n1
+    - kernel version is >= 4.19
+1) Call `$ make` in the root directory. Ensure this results in a scsi2nvme.ko
+1) Call `$ insmod scsi2nvme.ko` to insert the kernel module.
+
+    Upon inserting, the SCSI mock device will be automatically attached. Once the device is attached, the SCSI subsystem will issue a set of commands. 
+
+    `$ lsscsi` should show the device with the name "NVMe &nbsp; &nbsp; &nbsp; &nbsp;"
+
+### Issuing commands ###
+Issue commands to the SCSI device as such:
+
+Read: 
+>`$ sudo dd if=/dev/sdb bs=$BLOCK_SIZE count=$NUM_BLOCKS skip=STARTING_BLOCK_ADDRESS | hexdump -C`
+
+Write:
+>`$ sudo dd of=/dev/sdb bs=$BLOCK_SIZE count=$NUM_BLOCKS skip=STARTING_BLOCK_ADDRESS | hexdump -C`
```

Comment:
Should we be mentioning what exactly to write to the disk here?

Reply:
that's a good point!